src/components/AdvanceAnalyticsV2/data/utils.js

```diff
@@ -90,16 +89,24 @@ e.g. 3 would mean 3-day moving average.
 */
 const calculateMovingAverage = (timeSeriesData, countKey, period) => {
```

Comment:
[curious/consideration] Do you feel there are tests adequately covering the logic in this function? I'm still somewhat concerned about including these changes in a PR primarily about migrating to `lodash-es` given it's a refactor of a reasonably non-trival logic code path.

For example, iff there's a bug in this refactored logic, it'd be nice to be able to revert/rollback just these changes vs. all of the `lodash-es` changes. Maybe split these changes out into their own distinct PR and tag some folks from markhors on it given they primarily own this area and may have additional insights?

Reply:
I can revert the changes related to this function back to its original usage to instead leverage `.reduce` to handle summation of the moving average vs `sum` and open a seperate PR/ticket that includes this function optimization.

The function was very indicative of a leetcode-esque problem with an already optimized solution so I felt compelled to correct it on the spot.